docs/design-proposals/non_evm_inbound_memo/2024_10_02_standard_memo_format.md

```diff
@@ -0,0 +1,131 @@
+<h1 style="text-align: center;">A proposed standard memo format for non-EVM chains</h1>
+<br>
+
+## The goal
+
+This doc is aimed to introduce a standard format of ZetaChain inbound `memo` across non-EVM chains. The `memo` in an inbound transaction is to carry both instructions and data, by which the cross-chain transaction to be processed as expected in the `zetacore`. A standard memo format helps to ensure consistent developer experience across EVM and non-EVM chains.
+<br><br>
+
+
+## The EVM gateway inbound events
+
+We want to mimic below `Deposit` and `Called` events in the EVM `gateway` contract for non-EVM chains. Typically, we at least want the following 3 pieces at the time of writing.
+
+- The `receiver` address in ZEVM.
+- The `payload` of the call.
+- The `RevertOptions`.
+
+```
+event Deposited(
+    address indexed sender,
+    address indexed receiver,
+    uint256 amount,
+    address asset,
+    bytes payload,
+    RevertOptions revertOptions
+);
+
+event Called(
+	   address indexed sender,
+	   address indexed receiver,
+	   bytes payload,
+	   RevertOptions revertOptions
+);
+```
+<br><br>
+
+
+## 1. The Memo Diagram
+
+![](layout.png)<br><br>
+
+
+## 2. The memo structure
+
+The raw-byte representation of a standard non-EVM memo contains `9` sections (2 mandatory + 7 optional). 
+
+`Section 0` is the `header`, which controls how the memo should be interpreted.
+
+`Section 1~8` are data sections that carry actual data attached to the memo.
+
+
+|                 | Section 0  | Section 1   | Section 2   | Section 3      | Section 4    | Section 5    | Section 6     | Section 7   | Section 8   |
+|-----------------|------------|-------------|-------------|----------------|--------------|--------------|---------------|-------------|-------------|
+| **Name**        | Header     | Receiver    | Payload     | RevertAddress  | CallOnRevert | AbortAddress | RevertMessage | reserved    | reserved    |
```

Comment:
Any possible way `CallOnRevert` and `RevertMessage` could ever be used in the future? These are used for smart contracts chain to handle revert

Reply:
EDIT: Actually make sense to keep if we want to have this format standardized across all chains using a memo

Reply:
It will be ignored by `Bitcoin` observer but it could be useful in contract based non-EVM chains (e.g. `Solana` or `TON`) in future.

Reply:
The `CallOnRevert` and `RevertMessage` will be not used currently by Bitcoin. Let's leave the options consistent with EVM chains. The fields can be easily reused (or repurposed) if needed in future.